### PR TITLE
Premint - no operator overloading

### PR DESCRIPTION
**Pull request by oveddan**
* premint v2 - dont overload external operators, have specific names for each function
* premint validation methods now just take the contract address, allowing them to be used with existing contracts (not requiring contract creation config)
* extracted premint functionality to lib
* Defined interface for premint executor

### CODE REVIEW

**Comment by oveddan** [comment]
Current dependencies on/for this PR:
* main
  * **PR #278** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/278" target="_blank">Graphite</a> 
    * **PR #290** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/290" target="_blank">Graphite</a> 
      * **PR #291** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/291" target="_blank">Graphite</a> 
        * **PR #262** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/262" target="_blank">Graphite</a> 
          * **PR #294** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/294" target="_blank">Graphite</a> 
            * **PR #295** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/295" target="_blank">Graphite</a>  👈
            * **PR #289** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/289" target="_blank">Graphite</a> 
          * **PR #273** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/273" target="_blank">Graphite</a> 
            * **PR #288** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/288" target="_blank">Graphite</a> 
              * **PR #284** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/284" target="_blank">Graphite</a> 
            * **PR #280** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/280" target="_blank">Graphite</a> 
    * **PR #283** <a href="https://app.graphite.dev/github/pr/ourzora/zora-protocol/283" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ourzora/zora-protocol/295?utm_source=stack-comment).

File: packages/1155-contracts/src/delegation/ZoraCreator1155PremintExecutorImpl.sol [hunk 1]
```diff
@@ -212,35 +146,44 @@ contract ZoraCreator1155PremintExecutorImpl is
         return (true, ERC1155DelegationStorageV1(contractAddress).delegatedTokenId(uid));
     }
 
+    /// deprecated - use isValidSignatureV1
```
Comment (iainnash) [inline packages/1155-contracts/src/delegation/ZoraCreator1155PremintExecutorImpl.sol:149]: Can you make this a natspec so it shows up on etherscan?

can also add `@custom:deprecated use isValidSignatureV1 instead`

https://github.com/OpenZeppelin/openzeppelin-contracts/issues/3642

**Review by iainnash — APPROVED** [review]

**Review by IsabellaSmallcombe — APPROVED** [review]

File: packages/1155-contracts/src/delegation/ZoraCreator1155PremintExecutorImpl.sol [hunk 2]
```diff
@@ -90,30 +53,41 @@ contract ZoraCreator1155PremintExecutorImpl is
     /// @param signature Signature of the creator of the token, which must match the signer of the premint config, or have permission to create new tokens on the erc1155 contract if it's already been created
     /// @param quantityToMint How many tokens to mint to the executor of this transaction once the token is created
```
Comment (kulkarohan) [inline packages/1155-contracts/src/delegation/ZoraCreator1155PremintExecutorImpl.sol:54]: can update "executor" to "mintRecipient"

File: packages/1155-contracts/src/delegation/ZoraCreator1155PremintExecutorImpl.sol [hunk 3]
```diff
@@ -90,30 +53,41 @@ contract ZoraCreator1155PremintExecutorImpl is
...
     /// @param mintArguments Abi encoded additional mint arguments: including mintComment and mintReferral
```
Comment (kulkarohan) [inline packages/1155-contracts/src/delegation/ZoraCreator1155PremintExecutorImpl.sol:55]: can update "mintArguments" descrip

Comment (oveddan) [reply packages/1155-contracts/src/delegation/ZoraCreator1155PremintExecutorImpl.sol]: done

File: packages/1155-contracts/src/delegation/ZoraCreator1155PremintExecutorImpl.sol (hunk 2 above)
Comment (oveddan) [reply packages/1155-contracts/src/delegation/ZoraCreator1155PremintExecutorImpl.sol]: done

**Comment by oveddan** [comment]
merged into parent PR

**Pull request closed without merging**